### PR TITLE
Add compatibility for `transfer` to `continues_on` rename

### PR DESCRIPTION
**Pull request by msimberg**
If pika is older than 0.29.0, the adaptor is still called `transfer`. In that case we rename it to `continues_on` in an internal namespace. For 0.29.0 and later we simply alias pika's `continues_on` into an internal namespace.

### CODE REVIEW

**Comment by msimberg** [comment]
cscs-ci run

File: ci/common-ci.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ stages:
     reports:
       dotenv: build.env
   variables:
-    SPACK_SHA: 0905edf592752742eb4ddab3a528d3aee8f92930
+    SPACK_SHA: 1b9ba2b27d94a88fe384a07b4c8fd64f1cf23fb7
```
Comment (msimberg) [inline ci/common-ci.yml:38]: To do: change to commit on develop. This is from my fork.

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
cscs-ci run

**Comment by rasolca** [comment]
Not sure what happened in the failed CI. Usually clingo is bootstrapped.
Anyway I would rebase onto master to get the new CI pipeline.

**Review by RMeli — APPROVED** [review]
Apart from the odd CI failure, LGTM.

**Comment by msimberg** [comment]
cscs-ci run

**Review by albestro — APPROVED** [review]
LGTM

**Pull request merged**